### PR TITLE
natssysclient: Add system API client module

### PR DESCRIPTION
**Pull request by piotrpio**
This module exposes a system API client, used to request monitoring data from the server. Most of it is moved from here: https://github.com/piotrpio/nats-sys-client

Compared to the linked repo, there were some adjustments, most important being:
- context is used to handle timeouts
- module uses `RequestMany` from `natsext` when pinging the servers

Signed-off-by: Piotr Piotrowski [piotr@synadia.com](mailto:piotr@synadia.com)

### CODE REVIEW

File: natssysclient/api.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+	if s.opts.multiRequestInterval > 0 {
+		opts = append(opts, natsext.RequestManyStall(s.opts.multiRequestInterval))
+	}
+
+	if _, ok := ctx.Deadline(); !ok {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, DefaultRequestTimeout)
+		defer cancel()
+	}
+	iter, err := natsext.RequestMany(ctx, s.nc, subject, data, opts...)
```
Comment (Jarema) [inline natssysclient/api.go:128]: hm, maybe its worth wrapping the error with some contextual info?

File: natssysclient/api.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+		}
+		if initialTimeout <= 0 {
+			initialTimeout = interval
+		}
+		opts.multiRequestInterval = interval
+		return nil
+	}
+}
+
+// ServerCount sets the maximum number of servers to wait for the response from.
```
Comment (Jarema) [inline natssysclient/api.go:61]: I would improve the docs mentioning that this is not necessary most of the time, as the stall should be good enough.

File: natssysclient/api.go [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+		defer cancel()
+	}
+	iter, err := natsext.RequestMany(ctx, s.nc, subject, data, opts...)
+	if err != nil {
+		return nil, err
+	}
+	msgs := make([]*nats.Msg, 0)
+	for msg, err := range iter {
+		if err != nil {
+			return nil, err
```
Comment (Jarema) [inline natssysclient/api.go:135]: again - maybe worth wrapping the error?

**Review by Jarema — CHANGES_REQUESTED** [review]
Looks good. Just somes comments.

I would consider wrapping most errors with contextual info.

File: natssysclient/api.go (hunk 2 above)
Comment (ripienaar) [reply natssysclient/api.go:61]: Could also mention that the normal PING endpoint includes the overall cluster size as seen by the server. I often just ask my connected server for that and then use that expected size when making many requests in a row

**Review by ripienaar — APPROVED** [review]
Pretty cool I left just one comment. 

I am not usually a fan of pulling in server structs via a dependency but I hope we have a plan for keeping this current and we should clearly state what version of structs this matches

**Review by Jarema — APPROVED** [review]
LGTM

**Pull request merged**